composer.json

```diff
@@ -89,21 +90,25 @@
       "\"vendor/bin/phpcs\" --config-set installed_paths ../../../vendor/wp-coding-standards/wpcs,../../../vendor/lifterlms/lifterlms-cs,../../../vendor/phpcompatibility/php-compatibility,../../../vendor/phpcompatibility/phpcompatibility-paragonie,../../../vendor/phpcompatibility/phpcompatibility-wp",
       "\"vendor/bin/phpcs\" --config-set default_standard 'LifterLMS Core'"
     ],
-    "env": "vendor/bin/llms-env",
-    "env:setup": [
-      "vendor/bin/llms-env wp plugin activate lifterlms"
-    ],
+    "env": "\"vendor/bin/llms-env\"",
+    "env:setup": "\"vendor/bin/llms-env\" wp plugin activate lifterlms",
     "fix-cs": "\"vendor/bin/phpcbf\"",
-    "post-install-cmd": "composer config-cs",
-    "post-update-cmd": "composer config-cs",
-    "tests-install": [
-      "\"vendor/bin/llms-tests\" teardown llms_tests root password 127.0.0.1",
-      "\"vendor/bin/llms-tests\" install llms_tests root password 127.0.0.1"
+    "post-install-cmd": "@post-update-install-cmd",
+    "post-update-cmd": [
+      "@tests-remove",
+      "@post-update-install-cmd"
     ],
-    "tests-run": "\"vendor/bin/phpunit\"",
-    "phpcs": [
-      "\n\nWARNING: This command is deprecated, use `composer run check-cs` instead!\n\n",
-      "composer run check-cs"
-    ]
+    "post-update-install-cmd": [
+      "@tests-install",
```

Comment:
wait... I just went with a `composer install` after pulling this and got an error because the db was already installed.
I don't event think we should install tests on composer install to be honest, but if we want to do it then let's go with `@tests-reinstall`...
(so the same in the php8/composer.json above)

Reply:
Okay... this gets hairy...

In order to include the "patched" versions of files included in the `autoload-dev` in the composer config file we need to have the wp tests library installed.

If we don't install the test library then the `config-cs` command fails because the autoloader is attempting to load files that don't exist (yet)

Reply:
Ok that's fine, I see, so either we go with the reinstall or we add a check in our tests-install to skip the db creation if it already exists, but this is not critical I guess. <3

Reply:
I'll update the test install to skip if the db already exists, that seems better I think, right?

Reply:
Okay, I've made a bunch of changes based on your feedback:

Relevant updates from lifterlms-tests: https://github.com/gocodebox/lifterlms-tests/commit/0dc32075fae66ade4ea5d576ef39256a22f1facd

See recent commits for the rest

1. I added new composer scripts to make installing on php8 simpler
2. I've removed the tests installation from the post-install command
3. I've removed the necessity of having the class map in the autoload dev of the composer.json file

Now, during tests installation, if there are patched items in the wp tests library directory, they'll be copied directly into the phpunit directory, backups of the original versions will be made and automatically restored during test removal

This will simplify the steps and make switching PHP versions pretty simple, I hope.

Reply:
Works fine :D <3